### PR TITLE
DataFrame.to_records

### PR DESCRIPTION
**Pull request by shril**
@rxin, @ueshin Please have a look at my PR. #169 
Happy to help regarding my changes.

Reference Pandas Doc: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_records.html

### CODE REVIEW

**Comment by shril** [comment]
@rxin I am getting this exception for Pandas 0.23.4

```
UNEXPECTED EXCEPTION: TypeError("The pandas version [0.23.4] available does not support parameter 'column_dtypes' for function 'to_records'.",)
```

Should I skip doctest for it.

**Comment by shril** [comment]
@rxin, @ueshin I think I need some help here.

**Comment by garawalid** [comment]
@shril I suggest that you delete the parameter `column_dtypes` from `args` if  the pandas version is less than `0.23.4` 

```python
if LooseVersion(pd.__version__) < LooseVersion("0.24.0"):
    del args['column_dtypes'] 
```

File: databricks/koalas/frame.py [hunk 1]
```diff
@@ -1470,6 +1470,84 @@ def to_excel(self, excel_writer, sheet_name="Sheet1", na_rep="", float_format=No
         return validate_arguments_and_invoke_function(
             kdf.to_pandas(), self.to_excel, pd.DataFrame.to_excel, args)
 
+    def to_records(self, index=True, convert_datetime64=None,
+                   column_dtypes=None, index_dtypes=None):
+        """
+        Convert DataFrame to a NumPy record array.
+
+        Index will be included as the first field of the record array if
+        requested.
```
Comment (rxin) [inline databricks/koalas/frame.py]: also need a note warning users. see to_csv / to_json for examples

Comment (shril) [reply databricks/koalas/frame.py]: Thanks for mentioning.
Added.

**Comment by shril** [comment]
> @shril I suggest that you delete the parameter `column_dtypes` from `args` if the pandas version is less than `0.23.4`
> 
> ```python
> if LooseVersion(pd.__version__) < LooseVersion("0.24.0"):
>     del args['column_dtypes'] 
> ```

Thanks again @garawalid

**Comment by rxin** [comment]
@shril this doesn't merge cleanly anymore. Can you merge master into your branch so this can merge cleanly?

**Comment by HyukjinKwon** [comment]
Yea, many PRs are rapidly being merged. I will take a look as soon as it's merged to master.

**Comment by shril** [comment]
@rxin @HyukjinKwon I will look into this today.
Sorry for the delay.

**Comment by shril** [comment]
@rxin, @HyukjinKwon I have made the required changes. 
I apologize for the delay. Please have a look.
Happy to address any issues regarding my changes.

File: databricks/koalas/frame.py [hunk 2]
```diff
@@ -1572,6 +1572,90 @@ def to_excel(self, excel_writer, sheet_name="Sheet1", na_rep="", float_format=No
...
+
+        .. note:: This method should only be used if the resulting CSV is expected
```
Comment (HyukjinKwon) [inline databricks/koalas/frame.py]: I think the result isn't CSV

File: databricks/koalas/frame.py [hunk 3]
```diff
@@ -1572,6 +1572,90 @@ def to_excel(self, excel_writer, sheet_name="Sheet1", na_rep="", float_format=No
...
+        Examples
+        --------
+        >>> df = ks.DataFrame({'A': [1, 2], 'B': [0.5, 0.75]},
+        ...                   index=['a', 'b'])
+        >>> df
+           A     B
+        a  1  0.50
+        b  2  0.75
+
+        Data types (dtypes) are new in Pandas 0.24.0.
```
Comment (HyukjinKwon) [inline databricks/koalas/frame.py]: I think we don't have to mention this.

File: databricks/koalas/frame.py [hunk 4]
```diff
@@ -1572,6 +1572,90 @@ def to_excel(self, excel_writer, sheet_name="Sheet1", na_rep="", float_format=No
...
+        >>> df.to_records(column_dtypes={"A": "int32"}) # doctest: +SKIP
+        rec.array([('a', 1, 0.5 ), ('b', 2, 0.75)],
+                  dtype=[('index', 'O'), ('A', '<i4'), ('B', '<f8')])
+
+        Specification of dtype for index is new in Pandas 0.24.0.
+        Data types can also be specified for the index:
+
+        >>> df.to_records(index_dtypes="<S2") # doctest: +SKIP
+        rec.array([(b'a', 1, 0.5 ), (b'b', 2, 0.75)],
+                  dtype=[('index', 'S2'), ('A', '<i8'), ('B', '<f8')])
```
Comment (HyukjinKwon) [inline databricks/koalas/frame.py]: Are we checking different version of doc? Seems not matched - https://github.com/pandas-dev/pandas/blob/v0.24.2/pandas/core/frame.py#L1558-L1741

File: databricks/koalas/frame.py (hunk 2 above)
Comment (shril) [reply databricks/koalas/frame.py]: My bad. Correcting it.

File: databricks/koalas/frame.py (hunk 4 above)
Comment (shril) [reply databricks/koalas/frame.py]: @HyukjinKwon, I am having this issue that dtypes were added in Pandas 0.24.0. So the doc-tests are failing for the lower version of Pandas. So I am skipping the tests for them although I have 
pushed unit tests for every variation of the function.

File: databricks/koalas/frame.py (hunk 3 above)
Comment (shril) [reply databricks/koalas/frame.py]: Removed.

**Comment by HyukjinKwon** [comment]
Thanks, @shril

**Comment by shril** [comment]
You are most welcome @HyukjinKwon.

**Pull request merged**